### ISSUE TITLE
Pachyderm Dashboard

### ISSUE THREAD

**Issue by sjezewski** [pachyderm/pachyderm#718]
We've discussed this a lot internally, but I'm making an issue so that we don't lose track of some features we want to add. If we have a concrete use case we'd like the dashboard to address, let's record those details in this issue. This will help inform the design once we tackle it.

---

The dashboard will be deployed alongside a cluster, and will report on cluster usage / stats, and will be a debugging tool.

---

One thing that came up recently was looking at the # of client connections to the cluster. (Specifically when doing a ton of `PutFile()`s and perhaps not closing the connections)

As one of the metrics on the dashboard, we can report on the number of client connections.

**Comment by sjezewski** [comment]
Similarly, we might want to expose the number of current connections per pod. (So increment/decrement an atomic number when we start serving a request)

**Comment by sjezewski** [comment]
Once the block groupcache lands, we should have metrics reporting on:

- cache hits
- cache misses
- cache utilization (hit/miss ratio)
- per pod
- also mem consumption per pod?

**Comment by gabrielgrant** [comment]
There are actually some great ideas in here. Some of this seems like it's probably no longer relevant, but we should be giving this kind of feedback to help figure out how some of the more detailed tuning fields of a pipeline spec should be used. Specifically talking about `resource_spec` requests (and, soon, limits), `parallelism_spec`, `scale_down_threshold`, and `cache_size`

In general, we do an alright job of describing what these are in [the pipline spec docs](http://docs.pachyderm.io/en/latest/reference/pipeline_spec.html), but give basically zero guidance as to how/why to use them

@sjezewski I'm not sure which of the specific items listed in this issue are still relevant today. could you please create a meta-issue on the dash repo that migrates over the useful info (and link it back here for posterity)
